### ISSUE TITLE
"expiry-time" is zero to any PM page

### ISSUE THREAD

**Issue by gvp9000** [GwynethLlewelyn/post-local-storage#7]
See here ...
https://www.phpbb.com/community/viewtopic.php?t=2652829
Hope to find a solution ...
Actually its a phpBB problem.

As a workaround I put an "if" statement to bypass all PM pages.

```js
if (!key.includes("ucp.php")) {
      if (dateNow > expiry_time) {
      // We won't clear anything if the session already expired, so return.
      return;
      }
}
```
With the above code the local storage record is deleted after submitting a PM message.

**Comment by GwynethLlewelyn** [comment]
Hm. I guess I'm doing bad math on `expiry_time` when the page is a PM... looking into it!

**Comment by GwynethLlewelyn** [comment]
... or maybe the `<div id="expiry-time">...</div>` is not being properly injected into PM pages when the `posting_editor_options_prepend` gets triggered. Or, worse, `posting_editor_options_prepend` is _not_ triggered at all when writing a PM! Or... all the above! Yuck! Okay, I really need to take a look at all that...

**Comment by GwynethLlewelyn** [comment]
All right, I found the culprit! The event that assigns the session expiry time to the `<div id='expiry-time'>...</div>` wasn't being fired, or, rather, unlike what I thought, the triggers that were already configured are just fired for _posts_ but _not_ for PMs. I found one trigger for PMs which seems to work, so I'll consider this one closed for v1.2.1! 😸

**Comment by gvp9000** [comment]
Perfect !!!!!
